### PR TITLE
Upgrade the version of angle

### PR DESCRIPTION
**Pull request by sagudev**
Updates angle to [02755361e26d82768eb1d5f576145e19d7c265cd](https://hg.mozilla.org/mozilla-unified/file/02755361e26d82768eb1d5f576145e19d7c265cd/gfx/angle)

LibANGLE is not separate target anymore, but now we compile all gfx/angle/targets as separate libs (to be able to reuse as much as possible and avoid recompilation) and then link them into final targets (translator, libGLES, libEGL). We also need zlib (but only on egl feature).

Due to amount of changes (some of them also in public api), we need to do version bump (already included in PR).

Given how big this change is it would be good to test it on all consumers:
- [x] servo companion PR https://github.com/servo/servo/pull/30546
- [x] webrender companion: https://github.com/servo/webrender/pull/4824
- [ ] surfman (only has it for linking purposes)

### CODE REVIEW

**Comment by sagudev** [comment]
After reviewing https://bugzilla.mozilla.org/show_bug.cgi?id=1578576 I think only libGLES needs to depend on libANGLE stuff as libEGL depends on libGLES anyway.

**Comment by sagudev** [comment]
My plan is to do what is suggested in https://github.com/servo/mozangle/issues/33, so build once (static that is) and then convert it to dll.

**Comment by mrobinson** [comment]
I believe the change described in #33 was done in https://github.com/servo/mozangle/pull/58, but it seems like currently this PR undoes it?

**Comment by sagudev** [comment]
I am currently working on wr companion, linking works now, but some egl commands fail.

**Comment by sagudev** [comment]
I think this is ready for review! Companions PR are working so there shouldn't be any problems.

File: src/tests.rs
Comment (mrobinson) [inline src/tests.rs:50]: Looks like a stray println again?

File: patches/new-compilers.patch
Comment (mrobinson) [inline patches/new-compilers.patch:2]: Is this a patch that we are adding? Do you mind describing the purpose? I think "fix-build-on-newer-compilers.patch" was probably a better name.

**Review by mrobinson — CHANGES_REQUESTED** [review]
Nice work. I have a few comments, but this looks great in general.

File: build.rs
Comment (sagudev) [reply build.rs:163]: Not really as this is only printed if build script panics when executing, so it will help debugging.

File: build.rs
Comment (sagudev) [reply build.rs:171]: This comment is also on master. I think it should go to `fixup_path` function.

File: build.rs
Comment (sagudev) [reply build.rs:194]: This is set by libz-sys/rust, same is also used in mozjs. More here: https://doc.rust-lang.org/cargo/reference/build-script-examples.html#using-another-sys-crate

File: build.rs
Comment (sagudev) [reply build.rs:200]: It does, but this comment is just copy&paste from master.

File: build.rs
Comment (sagudev) [reply build.rs:251]: These two actually already have common field under `data.shared`, but I forgot to use it.

File: build.rs
Comment (sagudev) [reply build.rs:310]: In mozjs we also use glue term for cpp code that glues rust code with cpp code.

File: build.rs
Comment (sagudev) [reply build.rs:42]: This field is named after same build used in moz.build files.

File: patches/new-compilers.patch
Comment (sagudev) [reply patches/new-compilers.patch:2]: I thought this patch was not needed, but I needed to add part of it again so I did it lazily.

File: src/tests.rs
Comment (sagudev) [reply src/tests.rs]: All prints in tests.rs should indeed be removed, but I intended to keep prints in build.rs.

File: build.rs
Comment (mrobinson) [inline build.rs:355]: Hrm. This comment is a bit misleading because it suggests we are changing the "current working directory."

```suggestion
/// Make a path relative to the working directory that is used for the build.
fn fixup_path(path: &str) -> String {
```

File: generate_build_data.py
Comment (mrobinson) [inline generate_build_data.py:136]: `no_zlib` is so simple that it's useful to just leave it here:

```suggestion
    write_list(b"use_libs", map(lib_enum, filter(lambda s: "zlib" not in s, data["USE_LIBS"])), f)
```

**Review by mrobinson — APPROVED** [review]
Looks good, but with some changes below before landing.

I have one more question about build-time. Looking at the actions in GitHub it seems the build is now several minutes slower. Any idea what is causing that?

**Comment by sagudev** [comment]
> Looking at the actions in GitHub it seems the build is now several minutes slower. Any idea what is causing that?

Actually, first runs when I refactored build.rs to what it is now, build times were better than on master, but next few runs do not provide such great timings (newer runs contained only small fixups). As noted in servo Windows runners have varying performances.

Also given that new angle has a lot more code (the most important change being support for OpenGL), some changes in build time are expected (when WIP used minimal changed build.rs build times were 30min).

I do plan to investigate sccache support for followup, so we should get better build times.

**Comment by mrobinson** [comment]
> > Looking at the actions in GitHub it seems the build is now several minutes slower. Any idea what is causing that?
> 
> Actually, first runs when I refactored build.rs to what it is now, build times were better than on master, but next few runs do not provide such great timings (newer runs contained only small fixups). As noted in servo Windows runners have varying performances.
> 
> Also given that new angle has a lot more code (the most important change being support for OpenGL), some changes in build time are expected (when WIP used minimal changed build.rs build times were 30min).
> 
> I do plan to investigate sccache support for followup, so we should get better build times.

Great. Thanks for the information.

**Comment by sagudev** [comment]
I would still like to wait for review from @jdm given the size of PR.

**Comment by sagudev** [comment]
Jdm isn't able to review it, therefore we should just land it.

**Pull request merged**
